transformer_engine/jax/__init__.py

```diff
@@ -4,6 +4,7 @@
 """Transformer Engine bindings for JAX"""
 
 from . import flax
+from . import praxis
```

Comment:
We have Praxis as an optional dependency, but this import will fail if Praxis isn't installed.

Reply:
Yes, it is. Flax and Praxis are hard requirements for TE/JAX.
We import both `flax` ans `praxis` here is to align the usage of TE with other DLFWs in the style like 
```
import transformer.jax as te
te.flax.xxxx(...)
te.praxis.xxxx(...)
```

Without this, we cannot do like above usage.

Reply:
To avoid the hard requirement of `Flax` and `Praxis`, I propose to remove both `from . import flax` and `from . import praxis` and the trade off would be that users need to specific importing of NN library of TE/JAX. Like
```
import transformer.jax as te
te.fp8_autocast(...)

import transformer.jax.flax as te_flax
te_flax.xxxx()
import transformer.jax.praxis as te_praxis
te_praxis.xxxx()
```

Reply:
Update:
- Removed 'from . import praxis' in __init__.py.
    * Noted, keep 'from . import flax' for deprecated warning, will remove it in the near feature.
- Changed te.flax to te_flax in examples and README.rst.